### ISSUE TITLE
Is it possible to do a batch query on a BayesianNetwork object?

### ISSUE THREAD

**Issue by itskalvik** [pgmpy/pgmpy#1504]
### Subject of the issue
I have a BayesianNetwork model in which I want to make multiple queries. Doing it one at a time is relatively slow, and I wondered if we can do it in a batch-wise manner? I am using joblib for CPU parallelization, but is there a more efficient approach?

**Comment by ankurankan** [comment]
@kdkalvik No, unfortunately, the current implementation is limited to doing single queries at a time. I would suggest giving it a try without CPU parallelization as well as that can sometimes increase the run-time because of overheads.

**Comment by Giovannibriglia** [comment]
Any updates? :)

**Comment by ankurankan** [comment]
@Giovannibriglia This is unfortunately still not implemented. I do have a working algorithm but I haven't been able to finish the implementation yet: https://github.com/pgmpy/pgmpy/pull/1555. If you want to try it out, I have put a very basic working (works only with hard evidence) implementation here: https://gist.github.com/ankurankan/dc553ed89fbe11fff908b93e58bb9d2b
